homeassistant/components/media_player/ue_smart_radio.py

```diff
@@ -0,0 +1,207 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = "mdi:radio"
+URL = "http://decibel.logitechmusic.com/jsonrpc.js"
+
+SUPPORT_UE_SMART_RADIO = SUPPORT_PLAY | SUPPORT_PAUSE | SUPPORT_STOP | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_TURN_ON | \
+    SUPPORT_TURN_OFF | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE
+
+PLAYBACK_DICT = {"play": STATE_PLAYING,
+                 "pause": STATE_PAUSED,
+                 "stop": STATE_IDLE}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def send_request(payload, session):
+    """Send request to radio."""
+    try:
+        request = requests.post(URL,
+                                cookies={"sdi_squeezenetwork_session":
+                                         session},
+                                json=payload, timeout=5)
+    except requests.exceptions.Timeout:
+        _LOGGER.error("Timed out when sending request")
+    except requests.exceptions.ConnectionError:
+        _LOGGER.error("An error occurred while connecting")
+    else:
+        return request.json()
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Logitech UE Smart Radio platform."""
+    email = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    session_request = requests.post("https://www.uesmartradio.com/user/login",
+                                    data={"email": email, "password":
+                                          password})
+    session = session_request.cookies["sdi_squeezenetwork_session"]
+
+    player_request = send_request({"params": ["", ["serverstatus"]]}, session)
+    player_id = player_request["result"]["players_loop"][0]["playerid"]
+    player_name = player_request["result"]["players_loop"][0]["name"]
+
+    add_devices([UERadioDevice(session, player_id, player_name)])
+
+
+class UERadioDevice(MediaPlayerDevice):
+    """Representation of a Logitech UE Smart Radio device."""
+
+    def __init__(self, session, player_id, player_name):
+        """Initialize the Logitech UE Smart Radio device."""
+        self._session = session
+        self._player_id = player_id
+        self._name = player_name
+        self._state = None
+        self._volume = 0
+        self._last_volume = 0
+        self._media_title = None
+        self._media_artist = None
+        self._media_artwork_url = None
+
+    def send_command(self, command):
+        """Send command to radio."""
+        send_request({"method": "slim.request", "params":
+                      [self._player_id, command]}, self._session)
+
+    def update(self):
+        """Get the latest details from the device."""
+        request = send_request({
+            "method": "slim.request", "params":
+            [self._player_id, ["status", "-", 1,
+                               "tags:cgABbehldiqtyrSuoKLN"]]}, self._session)
```

Comment:
This contains quite some protocol specific info which we should extract into a new lib.